pkg/rulefmt/rulefmt.go

```diff
@@ -50,37 +51,67 @@ func (err *Error) Error() string {
 	} else if err.Err.node != nil {
 		return errors.Wrapf(err.Err.err, "%d:%d: group %q, rule %d, %q", err.Err.node.Line, err.Err.node.Column, err.Group, err.Rule, err.RuleName).Error()
 	}
+
 	return errors.Wrapf(err.Err.err, "group %q, rule %d, %q", err.Group, err.Rule, err.RuleName).Error()
 }
 
 // RuleGroups is a set of rule groups that are typically exposed in a file.
 type RuleGroups struct {
 	Groups []RuleGroup `yaml:"groups"`
+	nodes  ruleGroups
+}
+
+// RuleGroup is a list of sequentially evaluated recording and alerting rules.
+type RuleGroup struct {
+	Name     string         `yaml:"name"`
+	Interval model.Duration `yaml:"interval,omitempty"`
+	Rules    []RuleNode     `yaml:"rules"`
 }
 
+// RuleNode adds yaml.v3 layer to support line and column outputs for invalid rules.
```

Comment:
We don't need to mention yamlv3 here, describe what RuleNode represents.

Why do we need both RuleNode and Rule?

Reply:
`RuleNode` for yaml.v3 features but yes, it is not required to be public. Will change that to private since it is only used in the rulefmt.go.

Reply:
Sure. I have removed `Rule` and converted the `RuleNode` to `Rule` having the yaml properties from the v3 library. Everything should work now with just the Rule only.